modAionImpl/src/org/aion/zero/impl/sync/FastSyncManager.java

```diff
@@ -293,8 +302,25 @@ public BlocksWrapper takeFilteredBlocks(ByteArrayWrapper requiredHash, long requ
     }
 
     private void makeBlockRequests(ByteArrayWrapper requiredHash, long requiredLevel) {
-        // TODO: if the required hash is not among the known ones, request it from the network
-        // TODO: block requests should be made backwards from pivot
+        // make request for the needed hash
+        RequestBlocks request =
+                new RequestBlocks(requiredHash.getData(), BLOCKS_REQUEST_MAXIMUM_BATCH_SIZE, true);
+
+        // TODO: improve peer selection
         // TODO: request that level plus further blocks
+        INode peer = p2pMgr.getRandom();
```

Comment:
Check the best block of the picked peer can reduce the redundant request.

Reply:
As I mentioned in the PR description the `FastSyncManager` is still incomplete. The functionality in it is very basic. An optimized clean version will come after the receipts are also in the code base.